Bottom: b70b288cb02fd8e77c80f06128871e14eb4b4ec3
Top:    5d75ca02f120fa4a21fe9e2a3ba880d1546ce025
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:21:42 +0000

Refresh of improve-dump_poly-output

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 308519a..226b8ca 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2966,25 +2966,10 @@ poly_M_CheckInside (POLYAREA * p, Vector v0)
 int
 poly_ContourInContour (PLINE * poly, PLINE * inner)
 {
-  VNODE *pt;
   assert (poly != NULL);
   assert (inner != NULL);
-  if (cntrbox_inside (inner, poly)) {
-#if 1
-    pt = &inner->head;
-    /* HACK: Check all points on the contour being tested, because we don't
-     *       want to falsely return that two contours are inside each other
-     *       if they just touch at a few points.
-     */
-    do {
-      if (!poly_InsideContour (poly, pt->point))
-        return 0;
-    } while ((pt = pt->next) != &inner->head);
-    return 1;
-#else
+  if (cntrbox_inside (inner, poly))
     return poly_InsideContour (poly, inner->head.point);
-#endif
-  }
   return 0;
 }
